core/block_validator.go

```diff
@@ -34,18 +35,24 @@ const badBlockCacheExpire = 30 * time.Second
 //
 // BlockValidator implements Validator.
 type BlockValidator struct {
-	config *params.ChainConfig // Chain configuration options
-	bc     *BlockChain         // Canonical block chain
-	engine consensus.Engine    // Consensus engine used for validating
+	config          *params.ChainConfig // Chain configuration options
+	bc              *BlockChain         // Canonical block chain
+	engine          consensus.Engine    // Consensus engine used for validating
+	remoteValidator *remoteVerifyManager
 }
 
 // NewBlockValidator returns a new block validator which is safe for re-use
-func NewBlockValidator(config *params.ChainConfig, blockchain *BlockChain, engine consensus.Engine) *BlockValidator {
+func NewBlockValidator(config *params.ChainConfig, blockchain *BlockChain, engine consensus.Engine, mode VerifyMode, peers verifyPeers) *BlockValidator {
 	validator := &BlockValidator{
 		config: config,
 		engine: engine,
 		bc:     blockchain,
 	}
+	if mode.NeedRemoteVerify() {
+		log.Info("this node is a fast node with remote state verifier.")
```

Comment:
Kindly remind: I think this is debug log, please remove it before this PR been merged